syncr_frontend/frontend.py

```diff
@@ -776,15 +788,94 @@ def show_drops(drop_id=None, message=None):
 
     if message is not None:
         performed_action = {'description': message}
-        flash(message)
+        if not testing:
+            flash(message)
 
     # File Actions
-    if request.method == 'POST':
+    if not testing and request.method == 'POST':
         if request.form.get('type') == 'open_file':
             open_file_location('PUT PROPER LOCATION HERE')
 
-    return render_template(
-        'show_drops.html', selected=selected_drop, subscribed=subscribed_drops,
-        owned=owned_drops, action=performed_action, selec_act=curr_action,
-        versions=file_versions,
-    )
+    if not testing:
+        return render_template(
+            'show_drops.html',
+            selected=selected_drop,
+            subscribed=subscribed_drops,
+            owned=owned_drops,
+            action=performed_action,
+            selec_act=curr_action,
+            versions=file_versions,
+        )
+    else:
+        return {
+            'selected_drop': selected_drop,
+            'subscribed_drops': subscribed_drops,
+            'owned_drops': owned_drops,
+            'performed_action': performed_action,
+            'curr_action': curr_action,
+            'version': file_versions,
+        }
+
+
+class FrontendHook:
+
+    selected_drop = []
```

Comment:
these should be initialized in `__init__` as `self.<thing> = <value>` unless you mean these to be class variables